module/library/categories/helper.js

```diff
@@ -64,8 +66,22 @@ module.exports = class LibraryCategoriesHelper {
 					},
 				}
 
+				// Fetch the organization extension document of the loggedin user
+				let orgExtension = await orgExtensionQueries.orgExtenDocuments({
+					tenantId: userDetails.userInformation.tenantId,
+					orgId: userDetails.userInformation.organizationId,
+				})
+
+				if (!orgExtension || orgExtension.length === 0) {
+					orgExtension = null
+				} else {
+					orgExtension = orgExtension[0]
+				}
+
 				matchQuery['$match']['tenantId'] = userDetails.userInformation.tenantId
 
+				matchQuery = applyVisibilityConditions(matchQuery, orgExtension, userDetails)
```

Comment:
@Prajwal17Tunerlabs actually we can add comments which shows the match queries created for all three visibility cases